spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -218,32 +250,45 @@ private SpannerStatementQueryExecutor() {
 				keyParts.add(parentKeyParts.next());
 				tagNum++;
 			}
-			orJoiner.add(andJoiner.toString());
+			orParts.add(andJoiner.toString());
 		}
-		String cond = orJoiner.toString();
+		String keyCause = orParts.stream().collect(
+				() -> new StringJoiner(") OR (", "(", ")").setEmptyValue(""),
```

Comment:
Can we use `orParts.stream().map(s -> "("+s+")").collect(Collectors.joining(" OR "))` instead?
That would be shorter and more readable.

Reply:
I thought I found a way to not wrap an item by parentheses when the stream has a single value only. But I was wrong. I'll switch to your implementation, thanks.